.github/workflows/test_setup_tools.yml

```
@@ -0,0 +1,41 @@
+# NOTE: this workflow is now running on Dronecode / PX4 AWS account.
+# - If you want to keep the tests running in GitHub Actions you need to uncomment the "runs-on: ubuntu-latest" lines
+#   and comment the "runs-on: [runs-on,runner=..." lines.
+# - If you would like to duplicate this setup try setting up "RunsOn" on your own AWS account try https://runs-on.com
+
+name: Setup Tools Test (Ubuntu)
+
+on:
+  push:
+    branches:
+      - 'main'
+      - 'stable'
+      - 'beta'
+      - 'release/*'
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  ubuntu-run:
+    strategy:
+      fail-fast: false
+      matrix:
+        version: [22, 24]
+    # runs-on: ubuntu-latest
+    runs-on: [runs-on,runner=8cpu-linux-x64,"image=ubuntu${{ matrix.version }}-full-x64","run-id=${{ github.run_id }}"]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      # - name: ownership workaround
+      #   run: git config --system --add safe.directory '*'
+
```

The following step seems to fix the the failing builds.

```suggestion
      - name: Install pre-requisites
        run: |
            # Install ZeroMQ library
            sudo apt-get update -y
            sudo apt-get install -y libzmq3-dev libunwind-dev

            # Clone and install cppzmq
            git clone https://github.com/zeromq/cppzmq.git
            cd cppzmq
            mkdir build
            cd build
            cmake ..
            sudo make install
            cd ../..
```

Installing `libunwind-dev` seems to be required for both Ubuntu 22.04 and Ubuntu 24.04 but the `zmq` and `cppzmq` install seems to me to be only needed for Ubuntu 24.04. Unfortunately I didn't find out where ZMQ is actually used as I didn't find it directly in the PX4 code and I'm therefore unsure whether it makes sense to also include the ZMQ install in the `ubuntu.sh` script of if it's better to add it to the pipeline and then for example add an additional note in the documentation.

Regardless I hope this is useful in some way. 

Best regards,
Gilbert

---

Thanks, @TannerGilbert. I also couldn't track down where we use ZeroMQ; perhaps @dagar can help here.

Either way, instead of fixing it for CI, I added the dependencies to the **Ubuntu.sh** install script since our users will be running into these problems when they upgrade to 22.04 and 24.04